lib/Conversion/MatmulToArith/MatmulToArith.td

```diff
@@ -0,0 +1,24 @@
+#ifndef LIB_CONVERSION_MATMULTOARITH_MATMULTOARITH_TD_
+#define LIB_CONVERSION_MATMULTOARITH_MATMULTOARITH_TD_
+
+include "mlir/Pass/PassBase.td"
+
+def MatmulToArith : Pass<"matmul-to-arith"> {
```

Comment:
Why not `linalg-to-tensor-ext`? I feel that this pass should also support other linalg operators like convolution in the future.

Reply:
Also, why not start with linalg.matvec instead of linalg.matmul?

Reply:
I can rename it to linalg-to-tensor-ext, particularly for future-proofing reasons. We started with linalg.matmul because, in exploring a demo, we found that tosa would lower to linalg.matmul instead of linalg.matvec.

Reply:
tosa fully connected's will lower to matmul since most ML models will do have tensors like batch_size x channels

https://mlir.llvm.org/docs/Passes/#-linalg-fold-unit-extent-dims I tried this pass too but it won't apply, I was hopeful that maybe they would reduce to matvecs. 

matvecs would be good support to add later for sure though